src/main/java/teammates/ui/webapi/action/GetCoursesAction.java

```diff
@@ -17,17 +23,42 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
-        gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
+        AccountAttributes account = logic.getAccount(userInfo.getId());
```

Comment:
There is a potential NPE if `userInfo` is null (unregistered student)

Reply:
@rrtheonlyone I know this is very implementation-specific and the information is hidden, but if the min. auth level is `LOGGED_IN`, this block will not be reached if `userInfo` is null (because it means the user credential is equivalent to `PUBLIC`).

Reply:
Ahh okay understood! 👍 (didnt realise this)